### PR TITLE
Scala native multi-threading support

### LINKED ISSUE #9056: Enable multithreading for ZIO Native support
This will require ditching the versions of many structures copy/pasted from Scala.js and using true multi-threaded implementations, potentially from the JDK if they have Scala Native analogues.

### PR DESCRIPTION
**Pull request by kyri-petrou**
/fixes #9056
/claim #9056

Full disclosure, I've been working on enabling multi-threading support for Scala Native even before the bounty, but I thought to give #9117 a chance. After a very long number of review cycles, personally I think that PR requires too many changes to be in a a fully working state that it would take a significant amount of time to guide via reviews, so I ended up proceeding with opening a PR with what I've prepared since.

Anyhow, so this PR:
1. Moves all the JVM code that can be used by Scala Native to `/jvm-native`. This means that the ZIO runtime in Scala Native is now using the same ZScheduler as in JVM applications
2. Patches some parts here and there where we can't use the JVM code so that the classes used by native are thread-safe. The most annoying one is `java.util.concurrent.ConcurrentLinkedDeque`, I will try to followup with the Scala Native team for what will be required to add support for it. I'm guessing it shouldn't be too difficult since `java.util.concurrent.ConcurrentLinkedQueue` is already supported
3. Runs all tests annotated with `nonFlaky` both for JVM and Native
4. Runs all the test suites in parallel (PS: Native is running tests faster in some cases than the JVM 💡)
5. Removes the `scala-event-loop` dependency
6. Probably a bunch of other things, but in short, most of the Scala Native code is the same as the JVM, with the odd exceptions of extremely JVM-specific things (like Java streams, stream compression, etc.)

### CODE REVIEW

File: project/BuildHelper.scala [hunk 1]
```diff
@@ -242,7 +243,13 @@ object BuildHelper {
   )
 
   def nativeSettings = Seq(
-    nativeConfig ~= { _.withMultithreading(false) },
+    nativeConfig ~= { cfg =>
+      val os = System.getProperty("os.name").toLowerCase
+      // For some unknown reason, we can't run the test suites in debug mode on MacOS
```
Comment (kyri-petrou) [inline project/BuildHelper.scala:248]: Note to self: Need to followup with the Scala Native team and mention this in the release notes

File: project/BuildHelper.scala [hunk 2]
```diff
@@ -242,7 +243,13 @@ object BuildHelper {
...
+      if (os.contains("mac")) cfg.withMode(Mode.releaseFast)
+      else cfg
```
Comment (kyri-petrou) [inline project/BuildHelper.scala:250]: Note to self: Need to followup with the Scala Native team and mention this in the release notes

File: project/plugins.sbt [hunk 3]
```diff
@@ -12,7 +12,7 @@ addSbtPlugin("org.portable-scala"                % "sbt-scala-native-crossprojec
 addSbtPlugin("org.portable-scala"                % "sbt-scalajs-crossproject"      % "1.3.2")
 addSbtPlugin("org.scala-js"                      % "sbt-scalajs"                   % "1.16.0")
 addSbtPlugin("org.scalameta"                     % "sbt-mdoc"                      % "2.5.4")
-addSbtPlugin("org.scala-native"                  % "sbt-scala-native"              % "0.5.3")
+addSbtPlugin("org.scala-native"                  % "sbt-scala-native"              % "0.5.4")
```
Comment (kyri-petrou) [inline project/plugins.sbt:15]: I tried updating this to 0.5.5 but there seems to be a bug or something has changed in the latest release and we get an OOM error during CI. I suspect that it's due to changes in the default memory allocations in Scala Native. I've [followed up with the Scala Native team on Discord](https://discord.com/channels/632150470000902164/635668881951686686/1274987097048219649) to see what might be the issue

Comment (kyri-petrou) [reply project/plugins.sbt:15]: Seems indeed like a [Scala Native issue](https://github.com/scala-native/scala-native/issues/4032), unrelated to paralellism / changes in this PR 😮‍💨

**Review by sideeffffect — APPROVED** [review]

**Review by jdegoes — APPROVED** [review]

**Pull request merged**